src/rust/catloop/futures/accept.rs

```diff
@@ -125,9 +125,13 @@ impl Future for AcceptFuture {
                 duplex_pipe,
             } => {
                 if let Some(handle) = DuplexPipe::poll(&self_.catmem, *qt_close)? {
-                    check_connect_request(&self_.catmem, handle, *qt_close).expect("magic connect");
-                    debug!("connection accepted!");
-                    return Poll::Ready(Ok((*remote, duplex_pipe.clone())));
+                    match check_connect_request(&self_.catmem, handle, *qt_close) {
+                        Ok(_) => {
```

Comment:
If the server does not get a magic number back from the new pipe, it should probably close the pipe and clean it up, not continue to use it.